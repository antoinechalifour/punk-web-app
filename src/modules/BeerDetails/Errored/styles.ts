import styled from "styled-components";

import notFound from "./not-found.png";

export const Wrapper = styled.div`
  height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;

  h1 {
    font-size: 3rem;
    margin: 2rem auto;
  }

  a {
    padding: 1rem 2rem;
    margin: 2rem auto;
    border-radius: 5px;

    background: #5491f7;
    background: linear-gradient(to left, #9675ff, #5491f7);

    color: #fff;
  }
`;

export const Image = styled.img.attrs({
  src: notFound
})`
  max-width: 250px;
  margin-bottom: 2rem;
  opacity: 0.2;
`;

export const Message = styled.p`
  font-weight: 300;
`;
