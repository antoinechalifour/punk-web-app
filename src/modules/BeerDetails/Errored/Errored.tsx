import React from "react";

import { Wrapper, Image, Message } from "./styles";
import { Link } from "react-router-dom";

export interface ErroredProps {}

export const Errored: React.FunctionComponent<ErroredProps> = () => (
  <Wrapper>
    <Image />

    <h1>Gloops</h1>
    <Message>The beer was not found.</Message>

    <Link to="/">Take me back home</Link>
  </Wrapper>
);
