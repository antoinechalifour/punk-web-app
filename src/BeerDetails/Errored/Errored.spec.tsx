import { shallow } from "enzyme";
import React from "react";

import { Errored, ErroredProps } from "./Errored";

it("should match snapshot", () => {
  const props: ErroredProps = {};

  const wrapper = shallow(<Errored {...props} />);

  expect(wrapper).toMatchSnapshot();
});
