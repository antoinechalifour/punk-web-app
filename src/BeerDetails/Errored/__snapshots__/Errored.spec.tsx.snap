// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<styled.div>
  <styled.img />
  <h1>
    Gloops
  </h1>
  <styled.p>
    The beer was not found.
  </styled.p>
  <Link
    replace={false}
    to="/"
  >
    Take me back home
  </Link>
</styled.div>
`;
